### PR TITLE
FH-3748 - Change the cmd fhc policies to follow the V3 standard and a…

### PR DESCRIPTION
**Pull request by camilamacedo86**
JIRA: https://issues.jboss.org/browse/FH-3748

**Following the local changes**
- Move the cmd fhc admin-policies for V3 standard
- Change the name of the command to follow structure of the app
- Add unit tests
- Improve outputs 
- Fix the help and informations about this command
- Fix command to delete users from AuthPolicy which are not working currently.

**Steps to verify**
- `fhc admin policies list`
- `fhc admin policies read --id=<id>`
- `fhc admin policies update --id=<id> --type=<type> --checkUserExists=<true|false> --checkUserApproved=<true|false> --configurations=<configurations>`
- `fhc admin policies delete --guid=<guid>`
- `fhc admin policies users list --id=<id>`
- `fhc admin policies users add --guid=<guid> --user=<user>`
- `fhc admin policies users delete --guid=<guid> --user=<user>`
- `fhc admin policies create --id=<id> --type=<type> --configurations=<configurations> --checkUserExists=<true|false> --checkUserApproved=<true|false>`

**Following the local tests**

- `fhc admin policies list`

screen shot 2017-07-11 at 4 57 40 pm

screen shot 2017-07-11 at 5 21 00 pm

- `fhc admin policies read --id=<id>`

screen shot 2017-07-11 at 5 21 38 pm

- `fhc admin policies update --id=<id> --type=<type> --checkUserExists=<true|false> --checkUserApproved=<true|false> --configurations=<configurations>`

screen shot 2017-07-11 at 5 23 47 pm

- `fhc admin policies delete --guid=<guid>`

```
Camilas-MacBook-Pro:fh-fhc cmacedo$ ./bin/fhc.js admin policies delete h5d7wp24gy5o4t6if7awkhyh
Auth Policy deleted successfully
Camilas-MacBook-Pro:fh-fhc cmacedo$ 
```

- `fhc admin policies users list --id=<id>`

screen shot 2017-07-11 at 5 49 25 pm

- `fhc admin policies users add --guid=<guid> --user=<user>`

```
Camilas-MacBook-Pro:fh-fhc cmacedo$ ./bin/fhc.js admin policies users add rvcvdinhdmi463rgwzxq3pj4 camilamacedo86@gmail.com
User added into Auth Policy successfully
Camilas-MacBook-Pro:fh-fhc cmacedo$ ./bin/fhc.js admin policies users add rvcvdinhdmi463rgwzxq3pj4 camilamacedo86@gmail.com --json
{
  "status": "ok"
}
Camilas-MacBook-Pro:fh-fhc cmacedo$ 

``` 

- `fhc admin policies users delete --guid=<guid> --user=<user>`

```
Camilas-MacBook-Pro:fh-fhc cmacedo$ ./bin/fhc.js admin policies users delete rvcvdinhdmi463rgwzxq3pj4 camilamacedo86@gmail.com 
User deleted into Auth Policy successfully
Camilas-MacBook-Pro:fh-fhc cmacedo$ 
```

- fhc admin policies create --id=<id> --type=<type> --configurations=<configurations> --checkUserExists=<true|false> --checkUserApproved=<true|false>

screen shot 2017-07-11 at 6 53 00 pm

### CODE REVIEW

File: lib/cmd/fh3/admin/policies/create.js [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+        params.configurations = JSON.parse(params.configurations);
+      } catch (x) {
+        return cb(i18n._("Error parsing 'configurations' parameter: ") + util.inspect(x));
+      }
+    }
+    var data = {
+      policyId:params.id,
+      policyType:params.type,
+      configurations:params.configurations || {},
+      checkUserExists:params.checkUserExists || false,
```
Comment (fh-qube) [inline lib/cmd/fh3/admin/policies/create.js]: MINOR Remove the literal "false" boolean value. [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3AS1125)

File: lib/cmd/fh3/admin/policies/create.js [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+      checkUserApproved:params.checkUserApproved || false
```
Comment (fh-qube) [inline lib/cmd/fh3/admin/policies/create.js]: MINOR Remove the literal "false" boolean value. [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3AS1125)

**Comment by fh-qube** [comment]
SonarQube analysis reported 2 issues:
* MINOR 2 minor

Watch the comments in this conversation to review them.

File: lib/cmd/fh3/admin/policies/create.js [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+    if (params.type && !validPolicyType(params.type)) {
+      return cb(util.format(i18n._("'<type>' must be one of %s"), validPolicies.join(', ')));
+    }
+
+    var data = {
+      policyId:params.id,
+      policyType:params.type,
+      configurations: params.configurations,
+      checkUserExists:params.checkUserExists || false,
+      checkUserApproved:params.checkUserApproved || false,
```
Comment (fh-qube) [inline lib/cmd/fh3/admin/policies/create.js]: MINOR Remove the literal "false" boolean value. [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3AS1125)

File: lib/cmd/fh3/admin/policies/create.js [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+  'customCmd': function(params, cb) {
+    if (params.type && !validPolicyType(params.type)) {
+      return cb(util.format(i18n._("'<type>' must be one of %s"), validPolicies.join(', ')));
+    }
+
+    var data = {
+      policyId:params.id,
+      policyType:params.type,
+      configurations: params.configurations,
+      checkUserExists:params.checkUserExists || false,
```
Comment (fh-qube) [inline lib/cmd/fh3/admin/policies/create.js]: MINOR Remove the literal "false" boolean value. [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3AS1125)

**Comment by fh-qube** [comment]
SonarQube analysis reported 2 issues:
* MINOR 2 minor

Watch the comments in this conversation to review them.

File: lib/cmd/fh3/admin/policies/create.js [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+  'OAUTH1',
+  'OAUTH2',
+  'LDAP',
+  'OPENID',
+  'FEEDHENRY',
+  'MBAAS'
+];
+
+module.exports = {
+  'desc' : i18n._('Update a Auth Policy'),
```
Comment (pb82) [inline lib/cmd/fh3/admin/policies/create.js]: maybe just `Update Auth Policy`

File: lib/cmd/fh3/admin/policies/create.js [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+  'examples' :
+    [{
+      cmd : 'fhc admin policies create --id=<id> --type=<type> --configurations=<configurations> --checkUserExists=<true|false> --checkUserApproved=<true|false>' ,
+      desc : i18n._('Create the Auth Policy with the <id>,<type>,<checkUserExists>,<checkUserApproved> and <config>')
```
Comment (pb82) [inline lib/cmd/fh3/admin/policies/create.js]: `Create Auth Policy with <id> ...`

File: lib/cmd/fh3/admin/policies/create.js [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+    0 : 'id',
+    1 : 'type',
+    2 : 'configurations',
+    3 : 'checkUserExists',
+    4 : 'checkUserApproved'
+  },
+  'describe' : {
+    'id' : i18n._('The Auth Policy ID'),
+    'type' : i18n._('The type of the Auth Policy'),
+    'configurations' : i18n._('It is a json object corresponding to the policy type. E.g ')
```
Comment (pb82) [inline lib/cmd/fh3/admin/policies/create.js]: `JSON Object corresponding to the policy type, e.g.`

File: lib/cmd/fh3/admin/policies/create.js [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+    + i18n._('\n          for OAuth2: "{"clientId": "1234567890.apps.example.com",  "clientSecret": "Wfv8DQw80hhyaBqnW37x5R23", "provider": "GOOGLE"}"')
+    + i18n._('\n          for LDAP: "{"authmethod": "simple", "url": "ldap://foo.example.com:389, "dn": "ou=people,dc=example,dc=com", "dn_prefix": "cn", "provider": "LDAP"}')
+    + i18n._('\n                       authmethod can be one of: "simple", "DIGEST-MD5", "CRAM-MD5", or "GSSAPI"')
+    + i18n._('\n          for MBAAS: "{"provider":"MBAAS","mbaas":"mqijsw3tgpn6htwmbbg27jl6","authEndpoint":"/test","defaultEnvironment":"dev"}"'),
+    'checkUserExists' : i18n._('True for the Auth Policy check if exist the user and false to not check it'),
```
Comment (pb82) [inline lib/cmd/fh3/admin/policies/create.js]: Maybe just `<boolean> check user before updating`

File: lib/cmd/fh3/admin/policies/create.js [hunk 9]
```diff
@@ -0,0 +1,90 @@
...
+    'checkUserApproved' : i18n._('True for the Auth Policy check if the user is approved and false to not check it'),
```
Comment (pb82) [inline lib/cmd/fh3/admin/policies/create.js]: Maybe just `<boolean> check if user is approved before updating`

File: lib/cmd/fh3/admin/policies/create.js [hunk 10]
```diff
@@ -0,0 +1,90 @@
...
+  'customCmd': function(params, cb) {
+    if (params.type && !validPolicyType(params.type)) {
+      return cb(util.format(i18n._("'<type>' must be one of %s"), validPolicies.join(', ')));
+    }
+
+    var configurationsJSON;
+    try {
+      configurationsJSON = JSON.parse(params.configurations);
+    } catch (e) {
+      return cb(i18n._("Invalid a JSON Object for the 'configurations' parameter."));
```
Comment (pb82) [inline lib/cmd/fh3/admin/policies/create.js]: remote the `a`: `Invalid JSON object for the 'configurations' parameter`

File: lib/cmd/fh3/admin/policies/list.js [hunk 11]
```diff
@@ -0,0 +1,34 @@
...
+  },
+  'method' : 'get',
+  'postCmd': function(argv, response, cb) {
+    if (!argv.json && response.status === "ok") {
+      var headers = ['GUID', 'Policy Id','Type', 'checkUserApproved', 'checkUserExists'];
+      var fields = ['guid', 'policyId','policyType', 'checkUserApproved', 'checkUserExists'];
+      if (response.list.length > 0) {
+        response._table = common.createTableFromArray(headers, fields, response.list);
+      } else {
+        return cb(null,  i18n._('Not found Auth Policies for the target domain'));
```
Comment (pb82) [inline lib/cmd/fh3/admin/policies/list.js]: `No Auth Policies found for the target domain`

File: lib/cmd/fh3/admin/policies/read.js [hunk 12]
```diff
@@ -0,0 +1,35 @@
+/* globals i18n */
+var common = require("../../../../common");
+var fhreq = require("../../../../utils/request");
+
+module.exports = {
+  'desc' : i18n._('Read a Auth Policy'),
+  'examples' :
+    [{
+      cmd : 'fhc admin policies read --id=<id>',
+      desc : i18n._('Read the Auth Policy with the <id>')
```
Comment (pb82) [inline lib/cmd/fh3/admin/policies/read.js]: You can remove the `the`: `Read Auth Policy with <id>`

File: lib/cmd/fh3/admin/policies/update.js [hunk 13]
```diff
@@ -0,0 +1,101 @@
...
+  'OAUTH1',
+  'OAUTH2',
+  'LDAP',
+  'OPENID',
+  'FEEDHENRY',
+  'MBAAS'
+];
+
+module.exports = {
+  'desc' : i18n._('Update a Auth Policy'),
```
Comment (pb82) [inline lib/cmd/fh3/admin/policies/update.js]: You can remove the `a`: `Update Auth Policy`

File: lib/cmd/fh3/admin/policies/update.js [hunk 14]
```diff
@@ -0,0 +1,101 @@
...
+    0 : 'id',
+    1 : 'type',
+    2 : 'checkUserExists',
+    3 : 'checkUserApproved',
+    4 : 'config'
+  },
+  'describe' : {
+    'id' : i18n._('The Auth Policy ID'),
+    'type' : i18n._('The type of the Auth Policy'),
+    'checkUserExists' : i18n._('True for the Auth Policy check if exist the user and false to not check it'),
```
Comment (pb82) [inline lib/cmd/fh3/admin/policies/update.js]: see above comment

File: lib/cmd/fh3/admin/policies/update.js [hunk 15]
```diff
@@ -0,0 +1,101 @@
...
+    'checkUserApproved' : i18n._('True for the Auth Policy check if the user is approved and false to not check it'),
```
Comment (pb82) [inline lib/cmd/fh3/admin/policies/update.js]: see above comment

File: lib/cmd/fh3/admin/policies/update.js [hunk 16]
```diff
@@ -0,0 +1,101 @@
...
+    'configurations' : i18n._('It is a json object corresponding to the policy type. E.g ')
```
Comment (pb82) [inline lib/cmd/fh3/admin/policies/update.js]: see above comment

File: lib/cmd/fh3/admin/policies/users/add.js [hunk 17]
```diff
@@ -0,0 +1,36 @@
+/* globals i18n */
+var common = require("../../../../../common");
+var fhreq = require("../../../../../utils/request");
+
+module.exports = {
+  'desc' : i18n._('Add the User into a Auth Policy'),
+  'examples' :
+    [{
+      cmd : 'fhc admin policies users add --guid=<guid> --user=<user>',
+      desc : i18n._('Add the <user> into a Auth Policy with the <guid>')
```
Comment (pb82) [inline lib/cmd/fh3/admin/policies/users/add.js]: `Add <user> to Auth Policy with <guid>`

File: lib/cmd/fh3/admin/policies/users/delete.js [hunk 18]
```diff
@@ -0,0 +1,36 @@
+/* globals i18n */
+var common = require("../../../../../common");
+var fhreq = require("../../../../../utils/request");
+
+module.exports = {
+  'desc' : i18n._('Delete a User from a Auth Policy'),
```
Comment (pb82) [inline lib/cmd/fh3/admin/policies/users/delete.js]: `Delete user from Auth Policy`

File: lib/cmd/fh3/admin/policies/users/delete.js [hunk 19]
```diff
@@ -0,0 +1,36 @@
...
+  'examples' :
+    [{
+      cmd : 'fhc admin policies users delete --guid=<guid> --user=<user>',
+      desc : i18n._('Delete the <user> from a Auth Policy with the <guid>')
```
Comment (pb82) [inline lib/cmd/fh3/admin/policies/users/delete.js]: `Delete <user> from Auth Policy with <guid>`

File: lib/cmd/fh3/admin/policies/users/delete.js [hunk 20]
```diff
@@ -0,0 +1,36 @@
...
+    'user' : i18n._('Unique UserId into RHMAP'),
+    'json' : i18n._("Output in json format")
+  },
+  'customCmd': function(params, cb) {
+    common.doApiCall(fhreq.getFeedHenryUrl(), "/box/srv/1.1/admin/authpolicy/removeusers", {guid: params.guid, users: [params.user]}, i18n._("Error adding users to policy: "), function(err, data) {
+      if (err) {
+        return cb(err);
+      }
+      if (!params.json && data.status === "ok") {
+        return cb(null, i18n._('User deleted into Auth Policy successfully'));
```
Comment (pb82) [inline lib/cmd/fh3/admin/policies/users/delete.js]: Maybe better: `User successfully removed from Auth Policy`

File: lib/cmd/fh3/admin/policies/users/list.js [hunk 21]
```diff
@@ -0,0 +1,40 @@
+/* globals i18n */
+var common = require("../../../../../common");
+var fhc = require("../../../../../fhc");
+
+module.exports = {
+  'desc' : i18n._('Read the User of a Auth Policy'),
+  'examples' :
+    [{
+      cmd : 'fhc admin policies users list --id=<id>',
+      desc : i18n._('Read the users of a Auth Policy with the <id>')
```
Comment (pb82) [inline lib/cmd/fh3/admin/policies/users/list.js]: `Read user of Auth Policy with <id>`

File: lib/cmd/fh3/admin/policies/users/list.js [hunk 22]
```diff
@@ -0,0 +1,40 @@
...
+    }],
+  'demand' : ['id'],
+  'alias' : {
+    'id' : 'i',
+    'json' : 'j',
+    0 : 'id'
+  },
+  'describe' : {
+    'id' : i18n._('ID of the policy which you want to chek the users'),
```
Comment (pb82) [inline lib/cmd/fh3/admin/policies/users/list.js]: `ID of the Auth Policy where the user is assigned to`

File: lib/cmd/fh3/admin/policies/users/list.js [hunk 23]
```diff
@@ -0,0 +1,40 @@
...
+        return cb(err);
+      }
+      if (!params.json && data.status === "ok") {
+        if ( data.users.length > 0 ) {
+          var headers = ['GUID', 'Email','Name', 'userId'];
+          var fields = ['guid', 'email','name', 'userId'];
+          params._table = common.createTableFromArray(headers, fields, data.users);
+          return cb(null, params);
+        } else {
+          return cb(null, i18n._('Not found users associated to this Auth Policy'));
```
Comment (pb82) [inline lib/cmd/fh3/admin/policies/users/list.js]: `No users associated with this Auth Policy`

File: lib/cmd/fh3/admin/policies/create.js (hunk 5 above)
Comment (camilamacedo86) [inline lib/cmd/fh3/admin/policies/create.js]: Create Auth Policy - Done 👍

File: lib/cmd/fh3/admin/policies/create.js (hunk 6 above)
Comment (camilamacedo86) [inline lib/cmd/fh3/admin/policies/create.js]: Done 👍

File: lib/cmd/fh3/admin/policies/create.js (hunk 7 above)
Comment (camilamacedo86) [inline lib/cmd/fh3/admin/policies/create.js]: Done 👍

File: lib/cmd/fh3/admin/policies/create.js (hunk 8 above)
Comment (camilamacedo86) [inline lib/cmd/fh3/admin/policies/create.js]: Done 👍

File: lib/cmd/fh3/admin/policies/create.js (hunk 9 above)
Comment (camilamacedo86) [inline lib/cmd/fh3/admin/policies/create.js]: Done 👍

File: lib/cmd/fh3/admin/policies/create.js (hunk 10 above)
Comment (camilamacedo86) [inline lib/cmd/fh3/admin/policies/create.js]: Done 👍

File: lib/cmd/fh3/admin/policies/list.js (hunk 11 above)
Comment (camilamacedo86) [inline lib/cmd/fh3/admin/policies/list.js]: Done 👍

File: lib/cmd/fh3/admin/policies/read.js (hunk 12 above)
Comment (camilamacedo86) [inline lib/cmd/fh3/admin/policies/read.js]: Done 👍

File: lib/cmd/fh3/admin/policies/update.js (hunk 14 above)
Comment (camilamacedo86) [inline lib/cmd/fh3/admin/policies/update.js]: Done 👍

File: lib/cmd/fh3/admin/policies/update.js (hunk 15 above)
Comment (camilamacedo86) [inline lib/cmd/fh3/admin/policies/update.js]: Done 👍

File: lib/cmd/fh3/admin/policies/update.js (hunk 16 above)
Comment (camilamacedo86) [inline lib/cmd/fh3/admin/policies/update.js]: Done 👍

File: lib/cmd/fh3/admin/policies/users/add.js (hunk 17 above)
Comment (camilamacedo86) [inline lib/cmd/fh3/admin/policies/users/add.js]: Done 👍

File: lib/cmd/fh3/admin/policies/users/delete.js (hunk 18 above)
Comment (camilamacedo86) [inline lib/cmd/fh3/admin/policies/users/delete.js]: Done 👍

File: lib/cmd/fh3/admin/policies/users/delete.js (hunk 19 above)
Comment (camilamacedo86) [inline lib/cmd/fh3/admin/policies/users/delete.js]: Done 👍

File: lib/cmd/fh3/admin/policies/users/delete.js (hunk 20 above)
Comment (camilamacedo86) [inline lib/cmd/fh3/admin/policies/users/delete.js]: Done 👍

File: lib/cmd/fh3/admin/policies/users/list.js (hunk 21 above)
Comment (camilamacedo86) [inline lib/cmd/fh3/admin/policies/users/list.js]: Done 👍

File: lib/cmd/fh3/admin/policies/users/list.js (hunk 22 above)
Comment (camilamacedo86) [inline lib/cmd/fh3/admin/policies/users/list.js]: Done 👍

File: lib/cmd/fh3/admin/policies/users/list.js (hunk 23 above)
Comment (camilamacedo86) [inline lib/cmd/fh3/admin/policies/users/list.js]: Done 👍

**Pull request merged**